### PR TITLE
week 16 nutrition app  Viktoria and Ida

### PR DESCRIPTION
**Pull request by VickBe**
Link to my deployed project:https://optimistic-johnson-72f5fd.netlify.app/

My reflections on how this week's project turned out: I'm not proud if this project. I'm very happy we got something to show though! The api was like a bucket with  holes in it :)

Things I'd like to have clarified or explained in more detail: We didn't get the loader to work but decided to not spend to much time on it.

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
-};
+
+// const onDetected = (code) => {
+//   console.log(`Code: ${code}`);
+//   fetch(`https://world.openfoodfacts.org/api/v0/product/${code}.json`)
+//     .then((data) => data.json())
+//     .then((json) => {
+//       console.log(json);
+//     });
+// };
```
[the remark this answers is not in the document]
Comment (VickBe) [reply code/src/App.js]: Done

File: code/src/components/ScanBarcode.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import { BarcodeScanner } from './BarcodeScanner'
+import { fetchProduct } from 'reducers/productStore'
+import { useDispatch } from "react-redux"
+import styled from 'styled-components'
+
+
+const ButtonText = styled.p`
+    font-size:48px;
+`
```
[the remark this answers is not in the document]
Comment (VickBe) [reply code/src/components/ScanBarcode.js:10]: Done

File: code/src/components/BarcodeScanner.js [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+const Instruction = styled.p`
+    font-size:23px;
+    font-weigth:bold;
+    margin-top:-530px;
+    color:#004445;
+    @media (max-width: 375px){
+      font-size:32px;
+      text-align:center;
+    }
+`
```
[the remark this answers is not in the document]
Comment (VickBe) [reply code/src/components/BarcodeScanner.js]: Done

File: code/src/components/ProductList.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  if (!item) return null
+
+
+
+  return (
+    <ProductContainer>
+      {item.product && (<PrductList><h1>Allergens:</h1><li>
+        {item.product.allergens_hierarchy.map((allergen) =>
+          (allergen.replace(/\w+:/, ' ').replace('-', ' ')))}</li></PrductList>)
+      }
```
[the remark this answers is not in the document]
Comment (VickBe) [reply code/src/components/ProductList.js]: Done

File: code/src/components/BarcodeScanner.js [hunk 5]
```diff
@@ -23,23 +47,28 @@ export const BarcodeScanner = ({ className, onDetected }) => {
...
-    };
+      Quagga.stop()
+    }
   }, []);
 
   return (
     <>
-      {initializing && <div>Starting camera...</div>}
+    <CameraContainer>
+      {initializing && <div><Loading /></div>}
```
[the remark this answers is not in the document]
Comment (VickBe) [reply code/src/components/BarcodeScanner.js]: I'm not sure if I'm following.  But mounting it in App.js at least made it visiable . I think...it's so quick I'm not sure!!

File: code/src/components/ScanBarcode.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  return (
+    <>
+      {!showScanner && (
+        <Container>
+        <ShowScanner type="button" onClick={() => setShowScanner(true)}>
+        <ButtonContainer><Emojis>click→📷</Emojis></ButtonContainer> 
+        </ShowScanner>
+        </Container>
```
[the remark this answers is not in the document]
Comment (VickBe) [reply code/src/components/ScanBarcode.js]: Done

**Comment by VickBe** [comment]
Thank You so much! Very good comments!!

**Comment by HIPPIEKICK** [comment]
You have a bug now, it's a loop in the `ProductList.js`. I think it might be because your styled component's name is `PrductList` and now you're mounting `ProductList` inside of the `ProductList` component. Also, have another look at how to check if an array is empty, I commented on it last time. And please don't close the pull requests until it's approved by a teacher 😇

**Comment by VickBe** [comment]
Ok! I thought that was the way to "resend" it. Sorry!

Den fre 15 maj 2020 kl 14:06 skrev Matilda Arvidsson <
notifications@github.com>:

**Pull request closed without merging**